flytekit/deck/deck.py

```diff
@@ -174,3 +174,39 @@ def get_deck_template() -> "Template":
         autoescape=select_autoescape(enabled_extensions=("html",)),
     )
     return env.get_template("template.html")
+
+
+class PythonDependencyDeck(Deck):
+    """
+    PythonDependencyDeck is a deck that contains information about the library name and version.
+    """
+
+    def __init__(self, name: Optional[str] = "Python Dependency", html: Optional[str] = ""):
+        super().__init__(name, html)
+
+    @property
+    def html(self) -> str:
+        try:
+            from flytekitplugins.deck.renderer import TableRenderer
+        except ImportError:
+            warning_info = "Plugin 'flytekit-deck-standard' is not installed. To display python dependency, install the plugin in the image."
+            logger.warning(warning_info)
+            return warning_info
+
+        import subprocess
+
+        import pandas as pd
+
+        try:
+            installed_packages = subprocess.check_output(["pip", "freeze"]).decode().split("\n")
```

Comment:
To be safe, I usually like adding this:

```suggestion
            installed_packages = subprocess.check_output([sys.executable, "-m", "pip", "freeze"], text=True).split("\n")
```

Also `text=True` handles the encoding.

I think this is okay for pip packages, but `pip freeze` will not show the non-python dependencies for conda users.